portfolio/src/main/webapp/script.js

```diff
@@ -29,8 +29,40 @@ hideables.forEach(hideable => {
   h3.addEventListener('click', () => ul.classList.toggle("hidden"));
 });
 
-async function getGreeting() {
-    const response = await fetch("/data");
-    const quote = await response.text();
-    document.getElementById("quote-container").innerHTML = quote;
-}
\ No newline at end of file
+async function getComments() {
+  const response = await fetch("/data");
+  const text = await response.text();
+  const comments = JSON.parse(text);
+
+  document.getElementById("loading-text").remove();
+  
+  const fatherDiv = document.getElementById("comments-container");
+
+  comments.forEach((comment) => {
+    //create div component for comment
+    const comDiv = document.createElement('div');
+    comDiv.classList.add("comment");
+
+    //create name header
+    const h4 = document.createElement('h4');
+    h4.classList.add('comment-name-date');
+    const node1 = document.createTextNode(comment.name);
+    h4.appendChild(node1);
+    comDiv.appendChild(h4);
+
+    //create date header
+    const dateNode = document.createTextNode(`${comment.date !== undefined ? comment.date : ""}`);
+    const dateElement = document.createElement('p');
+    dateElement.classList.add("comment-name-date");
+    dateElement.appendChild(dateNode);
+    comDiv.appendChild(dateElement);
+
+    //create text paragraph
+    const txt = document.createElement('p');
+    const node2 = document.createTextNode(`${comment.text}`);
+    txt.appendChild(node2);
+    comDiv.appendChild(txt);
+
+    fatherDiv.appendChild(comDiv);
```

Comment:
In the review of Alicja's solution, Leon noted that touching the DOM multiple times might have some client-side performance implications. How about adding all the elements to a single element detached from the document and then, after it's finished, append that single element to the document?

Reply:
Thanks for the advice! Just commited the change.